### PR TITLE
feat(dropdown): add experimental dropdown component

### PR DESCRIPTION
**Pull request by tw15egan**
Closes https://github.com/IBM/carbon-components/issues/1215

Adds in experimental `Dropdown`, `MultiSelect`, `ComboBox`, `ListBox` components

#### Changelog

**New**

- Experimental dropdown and various extra states

#### Testing / Reviewing

Ensure `Dropdown` works properly in all states, the original drop-down is unaffected

cc @laurenmrice 
Here is the [staging link](http://carbon-dev-environment-exp-dropdown-wise-nyala.mybluemix.net/?nav=dropdown)

### CODE REVIEW

**Comment by laurenmrice** [comment]
@tw15egan 

Browser: Firefox  

For Up and Up(Light) Dropdowns:
- There is an empty space between the drawer and the main field. Is this what you intended or to keep them flush together so they are touching? This also occurs in Chrome and Safari.
screen shot 2018-10-18 at 10 14 16 am

- Once you select an option, the focus border is too high on the top and comes outside of the field.
screen shot 2018-10-18 at 10 14 27 am

For all Dropdowns:
- The spacing of the text between the divider lines is not completely equidistant. Spacing looks correct in Chrome and Safari.
screen shot 2018-10-18 at 10 27 14 am

**Review by alisonjoseph — APPROVED** [review]
Code LGTM! 🎉 

I did also notice the space above the dropdown on the "up" version

**Comment by tw15egan** [comment]
Whoops, missed the up variant 😅 

Updated and pushed up a new staging site:

http://carbon-dev-environment-exp-dropdown-wise-nyala.mybluemix.net/?nav=dropdown

**Comment by laurenmrice** [comment]
Yep looks good now ! 🤩👍

File: src/components/list-box/_list-box.scss [hunk 1]
```diff
@@ -291,3 +291,328 @@ $list-box-menu-width: rem(300px);
...
+    color: $text-01;
+    border-bottom: 1px solid $ui-01;
+
+    &::before,
+    &::after {
+      background-color: $ui-03;
+    }
+  }
+
+  .bx--list-box__menu-item .bx--checkbox-label {
```
Comment (joshblack) [inline src/components/list-box/_list-box.scss]: Oh no! Looks like there are some lingering `bx` classes from the previous implementation 😬

**Review by joshblack — CHANGES_REQUESTED** [review]
Looks great! Noticed a couple style quirks (potentially):

screen shot 2018-10-18 at 4 38 41 pm

In IE11, it looks like the background color bleeds down. This could be a remnant of the previous implementation, in which case might be better to file as a bug ticket that we could fix separately. Also noticed it in the disabled styles:

screen shot 2018-10-18 at 4 37 33 pm

Let me know what you think! Also had a similar note before as to whether we should do un-nesting with these components, or if we want to just leave them as is 👍

File: src/components/list-box/_list-box.scss (hunk 1 above)
Comment (asudoh) [reply src/components/list-box/_list-box.scss]: Let's see if GH "change suggestion" feature works! 😉 

```suggestion
  .#${prefix}--list-box__menu-item .#${prefix}--checkbox-label {
```

File: src/components/list-box/_list-box.scss [hunk 2]
```diff
@@ -291,3 +291,328 @@ $list-box-menu-width: rem(300px);
...
+    width: 100%;
+  }
+
+  .bx--list-box__menu-item .bx--checkbox-label-text {
```
Comment (asudoh) [inline src/components/list-box/_list-box.scss]:
```suggestion
  .#${prefix}--list-box__menu-item .#${prefix}--checkbox-label-text {
```

File: src/components/list-box/_list-box.scss [hunk 3]
```diff
@@ -291,3 +291,328 @@ $list-box-menu-width: rem(300px);
...
+    cursor: pointer;
+    color: $text-01;
+    transition: all $transition--expansion $carbon--standard-easing;
+
+    &:hover {
+      background-color: $ui-03;
+    }
+  }
+
+  .#{$prefix}--list-box .bx--text-input {
```
Comment (asudoh) [inline src/components/list-box/_list-box.scss]:
```suggestion
  .#{$prefix}--list-box .#${prefix}--text-input {
```

**Review by asudoh — APPROVED** [review]
LGTM 👍 from my perspective - Thanks @tw15egan!

**Comment by tw15egan** [comment]
@joshblack unnested _most_ selectors, kept things like `focus` and `hover` nested. Personal preference of mine 😄 

I'll make a separate issue for the IE11 issue, as I'm having trouble getting the environment up and running on my machine in IE 😬

**Comment by joshblack** [comment]
Sounds good @tw15egan! 🎉

**Review by joshblack — APPROVED** [review]

**Pull request merged**